hypothesis-python/src/hypothesis/control.py

```diff
@@ -125,7 +125,9 @@ def event(value: str) -> None:
 
 
 @deprecated_posargs
-def target(observation: float, *, label: str = "") -> None:
+def target(
+    observation: float, *, label: str = "", allow_outside_given_test: bool = False
+) -> None:
     """Calling this function with a ``float`` observation gives it feedback
```

Comment:
We'll need to document this new argument too - add the following paragraph around line 152 (just above the first `.. note::`)

```rst
``allow_outside_given_test=True`` disables the check for an enclosing ``@given``
test, which can be useful when defining e.g. a customised ``assert_almost_equal``
which uses ``target()`` when possible, but can also be used in non-property-based
tests.
```